ngraph/core/src/op/gather.cpp

```diff
@@ -128,6 +128,172 @@ shared_ptr<Node> op::v1::Gather::clone_with_new_inputs(const OutputVector& new_a
+    auto data_rank = data_pshape.rank();
+    auto indices_rank = indices_pshape.rank();
+    auto axis_rank = axis_pshape.rank();
+
+    if (axis_rank.is_static() && axis_pshape.is_static())
+    {
+        const auto axis_is_scalar = axis_rank.get_length() == 0;
+        const auto axis_has_one_elem =
+            axis_rank.get_length() == 1 && axis_pshape[0].get_length() == 1;
+        NODE_VALIDATION_CHECK(this,
+                              axis_is_scalar || axis_has_one_elem,
+                              "Axes input must be scalar or have 1 element (shape: ",
+                              axis_pshape,
+                              ").");
+    }
+    int64_t axis = get_axis();
+    NODE_VALIDATION_CHECK(this, axis != AXIS_NOT_SET_VALUE, "The axis is not specified");
+
+    int64_t batch_dims = get_batch_dims();
+    NODE_VALIDATION_CHECK(this,
+                          batch_dims <= axis,
+                          "The batch_dims <= axis. But instead got: batch_dims = ",
+                          batch_dims,
+                          ", axis = ",
+                          axis);
+    if (data_rank.is_static())
+    {
+        NODE_VALIDATION_CHECK(this,
+                              axis >= 0 && axis < data_rank.get_length(),
+                              "The axis must be => 0 and < data_rank. But instead got axis = ",
+                              axis,
+                              " data_rank = ",
+                              data_rank.get_length());
+    }
+    if (indices_rank.is_static())
+    {
+        NODE_VALIDATION_CHECK(
+            this,
+            batch_dims < indices_rank.get_length(),
+            "The batch_dims must be < indices_rank. But instead got: batch_dims = ",
+            batch_dims,
+            ", indices_rank = ",
+            indices_rank.get_length());
+    }
+
+    if (data_rank.is_static() && indices_rank.is_static())
+    {
+        auto out_rank =
+            data_pshape.rank().get_length() + indices_pshape.rank().get_length() - 1 - batch_dims;
+        std::vector<Dimension> result_dims(out_rank);
+        PartialShape output_pshape(result_dims);
+
+        // implementation of out_shape formula
+        // data.shape[:batch_dims] + data.shape[batch_dims:axis] + indices.shape[batch_dims:] +
+        // data.shape[axis + 1:]
+        int i = 0;
+        for (; i < batch_dims; i++)
+        {
+            NODE_VALIDATION_CHECK(this,
+                                  data_pshape[i].compatible(indices_pshape[i]),
+                                  "Shapes ",
+                                  data_pshape,
+                                  " and ",
+                                  indices_pshape,
+                                  " are not consistent. data and indices must have equal or "
+                                  "intersecting sizes until batch_dims");
+
+            output_pshape[i] = data_pshape[i] & indices_pshape[i];
+        }
+        for (; i < axis; i++)
```

Comment:
If the input with axes is not specified then `axis = std::numeric_limits<int64_t>::max()` and we will iterate in the for ages. Is this expected?

Reply:
Before we will get here If axis is not set we will throw an exception 
https://github.com/openvinotoolkit/openvino/blob/0226f65be71cb16aedcd1a95d1ab7e072799eb4d/ngraph/core/src/op/gather.cpp#L191

Reply:
OK